src_c/mixer.c

```diff
@@ -261,18 +261,45 @@ _format_view_to_audio(Py_buffer *view)
     return format;
 }
 
+static void
+_pg_push_mixer_event(int type, int code)
+{
+    pgEventObject *e;
+    PyObject *dict, *dictcode;
+    SDL_Event event;
+    PyGILState_STATE gstate = PyGILState_Ensure();
+
+    dict = PyDict_New();
+    if (dict) {
+        if (type >= PGE_USEREVENT && type < PG_NUMEVENTS) {
+            dictcode = PyInt_FromLong(code);
+            PyDict_SetItemString(dict, "code", dictcode);
+            Py_DECREF(dictcode);
+        }
+        e = (pgEventObject *)pgEvent_New2(type, dict);
+        Py_DECREF(dict);
+
+        if (e) {
+            pgEvent_FillUserEvent(e, &event);
+#if IS_SDLv1
+            if (SDL_PushEvent(&event) < 0)
+#else
+            if (SDL_PushEvent(&event) <= 0)
+#endif
+                Py_DECREF(dict);
+            Py_DECREF(e);
+        }
+    }
+    PyGILState_Release(gstate);
```

Comment:
Since all branches of endsound_callback use the gil now, maybe this is something that can happen once in endsound_callback?

Reply:
Yeah I did think of that. But then there was the possibility that the function just goes from start to end directly, in which case the GIL will be an extra overhead